copy/copy.go

```diff
@@ -92,6 +93,7 @@ type Options struct {
 	ReportWriter     io.Writer
 	SourceCtx        *types.SystemContext
 	DestinationCtx   *types.SystemContext
+	LayerCopyHook    func(types.BlobInfo) bool
```

Comment:
This should really have a more descriptive name; `shouldLayerBeCopied` or something shorter to that effect; “hook” does not describe the function at all.